docs/devel/instrumentation.md

```diff
@@ -0,0 +1,36 @@
+Instrumenting Kubernetes with a new metric
+===================
+
+The following is a step-by-step guide for adding a new metric to the Kubernetes code base.
+
+We use the Prometheus monitoring system's golang client library for instrumenting our code. Once you've picked out a file that you want to add a metric to, you should:
+
+1. Import "github.com/prometheus/client_golang/prometheus".
+
+2. Create a top-level var to define the metric. For this, you have to:
+  1. Pick the type of metric. Use a Gauge for things you want to set to a particular value, a Counter for things you want to increment, or a Histogram or Summary for histograms/distributions of values (typically for latency). Histograms are better if you're going to aggregate the values across jobs, while summaries are better if you just want the job to give you a useful summary of the values.
+  2. Give the metric a name and description.
+  3. Pick whether you want to distinguish different categories of things using labels on the metric. If so, add "Vec" to the name of the type of metric you want and add a slice of the label names to the definition.
+
+   https://github.com/GoogleCloudPlatform/kubernetes/blob/cd3299307d44665564e1a5c77d0daa0286603ff5/pkg/apiserver/apiserver.go#L53
```

Comment:
The line numbers won't change because I linked to them at a particular commit hash rather than at HEAD. But if you still think inlining code snippets is better, I can do that.